### PR TITLE
Add progressbar support to most download actions

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1597387

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
I am only hesitant about the module name `snapcraft.internal.network` luckily it is `internal` but it makes me think this should be `snapcraft.internal.indicators` to later on move indicator ui stuff here.

File: snapcraft/internal/network.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    if total_length:
+        progress_bar = ProgressBar(
+            widgets=[message,
+                     Bar(marker='=', left='[', right=']'),
+                     ' ', Percentage()],
+            maxval=total_length)
+    else:
+        progress_bar = ProgressBar(
+            widgets=[message, AnimatedMarker()],
+            maxval=UnknownLength)
```
Comment (kyrofa) [inline snapcraft/internal/network.py]: I'm still grumpy about this.

File: snapcraft/internal/parts.py [hunk 2]
```diff
@@ -67,31 +61,10 @@ def execute(self):
             return
         self._request.raise_for_status()
 
-        self._download()
+        download_requests_stream(self._request, self.parts_yaml,
+                                 'Downloading parts list')
```
Comment (kyrofa) [inline snapcraft/internal/parts.py]: Yeah, this is really nice.

**Comment by kyrofa** [comment]
Big :+1: here, nice work. I agree on the module name though-- indicators is better.

**Comment by josepht** [comment]
LGTM

**Comment by sergiusens** [comment]
Just ftr, all good with this branch still :wink:

```
======================================================================
ERROR: demos_tests.test_shout.ShoutTestCase.test_shout
----------------------------------------------------------------------
output: {{{b''}}}

Traceback (most recent call last):
  File "/tmp/adt-run.XawG72/build.6SZ/real-tree/snaps_tests/demos_tests/test_shout.py", line 25, in test_shout
    self.build_snap(self.snap_content_dir)
  File "/tmp/adt-run.XawG72/build.6SZ/real-tree/snaps_tests/__init__.py", line 159, in build_snap
    self._clean(working_dir)
  File "/tmp/adt-run.XawG72/build.6SZ/real-tree/snaps_tests/__init__.py", line 164, in _clean
    self._run_command(command, project_dir)
  File "/tmp/adt-run.XawG72/build.6SZ/real-tree/snaps_tests/__init__.py", line 187, in _run_command
    'Expected output {!r} not found.'.format(expect)) from None
snaps_tests.CommandError: Error running command 'snapcraft clean' in 'demos/shout'. Expected output <class 'pexpect.exceptions.EOF'> not found.
```

**Pull request merged**